### PR TITLE
Add Support For AWS Dex Basic-Authentication

### PR DESCRIPTION
**Pull request by PatrickXYS**
It's for internal review.

### CODE REVIEW

File: kfdef/source/master/kustomization.yaml [hunk 1]
```diff
@@ -5,6 +5,7 @@ resources:
 - kfctl_anthos.yaml
 - kfctl_aws.yaml
 - kfctl_aws_cognito.yaml
+- kfctl_aws_dex.yaml
```
Comment (Jeffwan) [inline kfdef/source/master/kustomization.yaml]: I think our plan is to fit dex into kfctl_aws.yaml? isn't it?

File: kfdef/kfctl_aws_dex.v1.0.0.yaml [hunk 2]
```diff
@@ -0,0 +1,426 @@
+apiVersion: kfdef.apps.kubeflow.org/v1
```
Comment (Jeffwan) [inline kfdef/kfctl_aws_dex.v1.0.0.yaml]: Don't need to generate for v1.0.0 - v1.0.2. They are all historical versions

File: kfdef/source/master/kustomization.yaml (hunk 1 above)
Comment (PatrickXYS) [reply kfdef/source/master/kustomization.yaml]: Right, fixed in the new revision.

File: kfdef/kfctl_aws_dex.v1.0.0.yaml (hunk 2 above)
Comment (PatrickXYS) [reply kfdef/kfctl_aws_dex.v1.0.0.yaml]: Fixed.

**Pull request closed without merging**